### PR TITLE
Fix #408: Response.age is semantically a timedelta, not a datetime

### LINKED ISSUE #408: Response.age mishandled as datetime instead of int (or timedelta)
Environment:

```
Flask==0.10.1
Werkzeug==0.9.1
```

Test case:

``` py
#/usr/bin/env python2.7
from flask import Flask, Response

app = Flask(__name__)

@app.route('/')
def root():
    resp = Response()
    resp.age = 42
    return resp

if __name__ == "__main__":
    client = app.test_client()
    resp = client.get('/')
    print "age property:", resp.age
    print "Age header:", resp.headers['Age']
```

Result:

```
age property: 1970-01-01 00:00:42
Age header: Thu, 01 Jan 1970 00:00:42 GMT
```

Expected `42` instead in both cases (though personally I'd optimally prefer `timedelta(seconds=42)` in the former case).
[The HTTP spec says (in part; emphasis added):](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html)

> The Age response-header field conveys the sender's estimate of the **amount of time** since the response (or its revalidation) was generated at the origin server.
> `age-value = delta-seconds`
> Age values are **non-negative decimal integers**, representing time in seconds.

### PR DESCRIPTION
**Pull request by cvrebert**
First time I've tried to contribute to Werkzeug; criticism is welcome.
Fixes #408.

### CODE REVIEW

**Comment by untitaker** [comment]
FWIW, I've decided to delay this to 1.0, because we can cleanly break the API in that release. The patch itself seems to be fine and can be applied as-is.

**Comment by davidism** [comment]
@untitaker can we re-schedule this to 0.13 since it's been 2 years since 1.0 was added? I think this is ok for a feature release.

**Comment by untitaker** [comment]
Sure, but technically feature releases don't allow breakage at all. But 0.13 is not a feature release (pre-1.0 different rules apply in semver)

On 12 April 2017 17:12:03 CEST, David Lord <notifications@github.com> wrote:
>@untitaker can we re-schedule this to 0.13 since it's been 2 years
>since 1.0 was added? I think this is ok for a feature release.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pallets/werkzeug/pull/414#issuecomment-293609297

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by davidism** [comment]
Ah, I didn't know we were following semver, I just figured we were ok making big changes in 0.x releases. I'll merge this into master then.

**Comment by untitaker** [comment]
Pre-1.0 we would comply with semver anyway because semver allows for a lot at this stage.

On 12 April 2017 17:14:53 CEST, David Lord <notifications@github.com> wrote:
>Ah, I didn't know we were following semver, I just figured we were ok
>making big changes in 0.x releases. I'll merge this into master then.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pallets/werkzeug/pull/414#issuecomment-293610281

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: werkzeug/http.py [hunk 1]
```diff
@@ -729,6 +729,49 @@ def http_date(timestamp=None):
...
 
+def parse_age(value=None):
+    """Parses a base-10 integer count of seconds into a timedelta.
+
+    If parsing fails, the return value is `None`.
+
+    :param value: a string consisting of an integer represented in base-10
+    :return: a :class:`datetime.timedelta` object or `None`.
+    """
+    if not value:
```
Comment (davidism) [inline werkzeug/http.py]: Should be `not None`, since `0` is a valid age.

File: werkzeug/http.py [hunk 2]
```diff
@@ -729,6 +729,49 @@ def http_date(timestamp=None):
...
+                if the age is unknown, `None` (default).
+    """
+    if age is None:
+        return
+    if isinstance(age, timedelta):
+        # do the equivalent of Python 2.7's timedelta.total_seconds(),
+        # but disregarding fractional seconds
+        age = age.seconds + (age.days * 24 * 3600)
+
+    age = int(age)
```
Comment (davidism) [inline werkzeug/http.py]: `int` is fine.

**Comment by davidism** [comment]
#1104 rebases this against current master so that it can be cleanly merged.

**Pull request closed without merging**